src/structures/GuildEmoji.js

```diff
@@ -16,12 +16,7 @@ class GuildEmoji extends Emoji {
      */
     this.guild = guild;
 
-    /**
-     * A collection of roles this emoji is active for (empty if all), mapped by role ID
-     * @type {GuildEmojiRoleStore<Snowflake, Role>}
-     */
-    this.roles = new GuildEmojiRoleStore(this);
-
+    this._roles = null;
```

Comment:
Wouldn't iterating over `GuildEmoji#roles` before patching throw an error (Cannot read property 'includes' of null) because `_roles` is initialized with `null`?
This is more of an edge case, but could be avoided by initializing with an empty array.